doc/guides/dts/index.rst

```diff
@@ -346,44 +346,534 @@ Below is sample header content generated for this node, in
+enabled node:
+
+.. code-block:: none
+
+   #define DT_COMPAT_<compatible> 1
+
+An existence flag is also written for all enabled nodes that matched some
+binding:
+
+.. code-block:: none
+
+   #define DT_INST_<instance-no.>_<compatible>    1
+
+For the example ``sim@40047000`` node above, assuming the node is the first
+node with ``compatible = "nxp,kinetis-sim"``, the macros look like this:
+
+.. code-block:: c
+
+   #define DT_COMPAT_NXP_KINETIS_SIM    1
+   #define DT_INST_0_NXP_KINETIS_SIM    1
+
+Bus-related macros
+------------------
+
+These macros get generated for nodes that appear on buses (have ``on-bus:`` in
+their binding):
+
+.. code-block:: none
+
+   #define DT_<node>_BUS_NAME                "<bus-label>"
+   #define DT_<compatible>_BUS_<bus-name>    1
+
+``<bus-label>`` is taken from the ``label`` property on the bus node, which
+must exist. ``<bus-name>`` is the identifier for the bus as given in
+``on-bus:`` in the binding.
+
+.. _reg_macros:
+
+Macros generated from flash partitions
+--------------------------------------
+
+.. note::
+
+   This section only covers flash partitions. See :ref:`dt-alias-chosen` for
+   some other flash-related macros that get generated from devicetree, via
+   ``/chosen``.
+
+If a node has a name that looks like ``partition@<unit-address>``, it is
+assumed to represent a flash partition.
+
+Assume the devicetree has this:
+
+.. code-block:: none
+
+   flash@0 {
+   	...
+   	label = "foo-flash";
+
+   	partitions {
+   		...
+   		#address-cells = <1>;
+   		#size-cells = <1>;
+
+   		boot_partition: partition@0 {
+   			label = "mcuboot";
+   			reg = <0x00000000 0x00010000>;
+   			read-only;
+   		};
+   		slot0_partition: partition@10000 {
+   			label = "image-0";
+   			reg = <0x00010000 0x00020000
+   			       0x00040000 0x00010000>;
+   		};
+   		...
+   	}
+
+These macros then get generated:
+
+.. code-block:: c
+
+   #define DT_FLASH_AREA_MCUBOOT_ID           0
+   #define DT_FLASH_AREA_MCUBOOT_READ_ONLY    1
+   #define DT_FLASH_AREA_MCUBOOT_OFFSET_0     0x0
+   #define DT_FLASH_AREA_MCUBOOT_SIZE_0       0x10000
+   #define DT_FLASH_AREA_MCUBOOT_OFFSET       DT_FLASH_AREA_MCUBOOT_OFFSET_0
+   #define DT_FLASH_AREA_MCUBOOT_SIZE         DT_FLASH_AREA_MCUBOOT_SIZE_0
+   #define DT_FLASH_AREA_MCUBOOT_DEV          "foo-flash"
+
+   #define DT_FLASH_AREA_IMAGE_0_ID           0
+   #define DT_FLASH_AREA_IMAGE_0_READ_ONLY    1
+   #define DT_FLASH_AREA_IMAGE_0_OFFSET_0     0x10000
+   #define DT_FLASH_AREA_IMAGE_0_SIZE_0       0x20000
+   #define DT_FLASH_AREA_IMAGE_0_OFFSET_1     0x40000
+   #define DT_FLASH_AREA_IMAGE_0_SIZE_1       0x10000
+   #define DT_FLASH_AREA_IMAGE_0_OFFSET       DT_FLASH_AREA_IMAGE_0_OFFSET_0
+   #define DT_FLASH_AREA_IMAGE_0_SIZE         DT_FLASH_AREA_IMAGE_0_SIZE_0
+   #define DT_FLASH_AREA_IMAGE_0_DEV          "foo-flash"
+
+   /* Same macros, just with index instead of label */
+   #define DT_FLASH_AREA_0_ID           0
```

Comment:
Why is this in italics?

Reply:
It's not; GitHub just isn't rendering it properly. Are you seeing italics in the rendered HTML output?